tcell_compat.go

```diff
@@ -0,0 +1,524 @@
+	KeyCtrl3          = Key(tcell.KeyEscape)
+	KeyCtrl4          = Key(tcell.KeyCtrlBackslash)
+	KeyCtrl5          = Key(tcell.KeyCtrlRightSq)
+	KeyCtrl6          = Key(tcell.KeyCtrlCarat)
+	KeyCtrl7          = Key(tcell.KeyCtrlUnderscore)
+	KeyCtrlSlash      = Key(tcell.KeyCtrlUnderscore)
+	KeyCtrlRsqBracket = Key(tcell.KeyCtrlRightSq)
+	KeyCtrlBackslash  = Key(tcell.KeyCtrlBackslash)
+	KeyCtrlLsqBracket = Key(tcell.KeyCtrlLeftSq)
+)
+
+var (
+	lastMouseKey tcell.ButtonMask = tcell.ButtonNone
+	lastMouseMod tcell.ModMask    = tcell.ModNone
+)
+
+// Modifiers.
+const (
+	ModAlt  = Modifier(tcell.ModAlt)
+	ModNone = Modifier(0)
+	// ModCtrl doesn't work with keyboard keys. Use CtrlKey in Key and ModNone. This is for mouse clicks only (tcell.v1)
+	// ModCtrl = Modifier(tcell.ModCtrl)
+)
+
+func makeEvent(tev tcell.Event) Event {
+	switch tev := tev.(type) {
+	case *tcell.EventInterrupt:
+		return Event{Type: EventInterrupt}
+	case *tcell.EventResize:
+		w, h := tev.Size()
+		return Event{Type: EventResize, Width: w, Height: h}
+	case *tcell.EventKey:
+		k := tev.Key()
+		ch := rune(0)
+		if k == tcell.KeyRune {
+			k = 0 // if rune remove key (so it can match, for example spacebar)
+			ch = tev.Rune()
+		}
+		mod := tev.Modifiers()
+		// remove control modifier and setup special handling of ctrl+spacebar, etc.
+		if mod == tcell.ModCtrl && k == 0 && ch == 0 {
+			mod = 0
+			k = tcell.KeyCtrlSpace
+		} else if mod == tcell.ModCtrl {
+			mod = 0
+		}
+		return Event{
+			Type: EventKey,
+			Key:  Key(k),
+			Ch:   ch,
+			Mod:  Modifier(mod),
+		}
+	case *tcell.EventMouse:
+		x, y := tev.Position()
+		button := tev.Buttons()
+		mouseKey := MouseRelease
+		mouseMod := ModNone
+		// process mouse wheel
+		if button&tcell.WheelUp != 0 {
+			mouseKey = MouseWheelUp
+		}
+		if button&tcell.WheelDown != 0 {
+			mouseKey = MouseWheelDown
+		}
+		if button&tcell.WheelLeft != 0 {
+			mouseKey = MouseWheelLeft
+		}
+		if button&tcell.WheelRight != 0 {
+			mouseKey = MouseWheelRight
+		}
+
+		// process button events (not wheel events)
+		button &= tcell.ButtonMask(0xff)
+		if button != tcell.ButtonNone && lastMouseKey == tcell.ButtonNone {
+			lastMouseKey = button
+			lastMouseMod = tev.Modifiers()
+		}
+
+		switch tev.Buttons() {
+		case tcell.ButtonNone:
+			if lastMouseKey != tcell.ButtonNone {
+				switch lastMouseKey {
+				case tcell.ButtonPrimary:
+					mouseKey = MouseLeft
+				case tcell.ButtonSecondary:
+					mouseKey = MouseRight
+				case tcell.ButtonMiddle:
+					mouseKey = MouseMiddle
+				}
+				mouseMod = Modifier(lastMouseMod)
+				lastMouseMod = tcell.ModNone
+				lastMouseKey = tcell.ButtonNone
+			}
+		}
+
+		return Event{
+			Type:   EventMouse,
+			MouseX: x,
+			MouseY: y,
+			Key:    mouseKey,
+			Ch:     0,
+			Mod:    mouseMod,
+		}
+	default:
+		return Event{Type: EventNone}
+	}
+}
+
+// tcellPollEvent blocks until an event is ready, and then returns it.
+func tcellPollEvent() Event {
```

Comment:
This function is only used in one place (`gui.go:443`) so it's maybe better to just remove this function as it's only 2 lines of code

Reply:
moved to different function